src/compiler/parser.ts

```diff
@@ -3960,7 +3960,7 @@ namespace ts {
                     shorthandDeclaration.equalsToken = equalsToken;
                     shorthandDeclaration.objectAssignmentInitializer = allowInAnd(parseAssignmentExpressionOrHigher);
                 }
-                return finishNode(shorthandDeclaration);
+                return addJSDocComment(finishNode(shorthandDeclaration));

```

Comment:
I notice this function returns early if `tryParseAccessorDeclaration` is successful (or via `parseMethodDeclaration`@3943). Do we handle those cases, such as getters/setters? (Good tests to add also).